### PR TITLE
[Console] Add sigv4 signing to cluster api calls

### PR DESCRIPTION
**Pull request by mikaylathompson**
### Description
This is a stripped down version of #854 that focuses just on the python/console side of adding sigv4, instead of the CDK side (which will be pulled out and done as a separate PR).

### Issues Resolved
[[Console] Implement sigv4 signing for clusters](https://opensearch.atlassian.net/browse/MIGRATIONS-1759)

### Testing
Unit & manual

### Check List
- [x] New functionality includes testing
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/cluster.py [hunk 1]
```diff
@@ -103,6 +110,17 @@ def get_basic_auth_password(self) -> str:
...
 
+    def _get_sigv4_details(self, force_region=False) -> tuple[str, str]:
+        """Return the service signing name and region name. If force_region is true,
+        it will instantiate a boto3 session to guarantee that the region is not None.
+        This will fail if AWS credentials are not available.
+        """
+        assert self.auth_type == AuthMethod.SIGV4
+        if force_region:
+            session = boto3.session.Session()
+            return self.auth_details.get("service", "es"), self.auth_details.get("region", session.region_name)
```
Comment (AndreKurait) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/cluster.py:121]: nit: Can we initiate the boto3 session in such a way that it's only initiated if region is None?

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/cluster.py [hunk 2]
```diff
@@ -103,6 +110,17 @@ def get_basic_auth_password(self) -> str:
         password = client.get_secret_value(SecretId=self.auth_details["password_from_secret_arn"])
         return password["SecretString"]
 
+    def _get_sigv4_details(self, force_region=False) -> tuple[str, str]:
+        """Return the service signing name and region name. If force_region is true,
+        it will instantiate a boto3 session to guarantee that the region is not None.
+        This will fail if AWS credentials are not available.
```
Comment (AndreKurait) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/cluster.py:116]: From the unit tests, it seems this isn't actually the behavior, and instead will only fail if AWS Credentials are not available AND a region is not specified in auth details

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/Pipfile
Comment (AndreKurait) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/Pipfile:1]: Can you follow instructions here to update all Pipfile.locks? https://github.com/opensearch-project/opensearch-migrations/blob/4727d1ad140b4f0eb5e790fa9f8b35c1d05dfb32/TrafficCapture/dockerSolution/README.md?plain=1#L116

Comment (mikaylathompson) [reply TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/Pipfile:1]: Done

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/cluster.py [hunk 3]
```diff
@@ -116,7 +116,7 @@ def _get_sigv4_details(self, force_region=False) -> tuple[str, str]:
         This will fail if AWS credentials are not available.
         """
         assert self.auth_type == AuthMethod.SIGV4
-        if force_region:
+        if force_region and 'region' not in self.auth_details:
             session = boto3.session.Session()
             return self.auth_details.get("service", "es"), self.auth_details.get("region", session.region_name)
```
Comment (AndreKurait) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/cluster.py:121]: nit: would have been more readable to exclude `self.auth_details.get("region",`

**Review by AndreKurait — APPROVED** [review]
Thanks!

**Pull request merged**